### PR TITLE
Fix HttpClientRequestMessage.cs

### PR DESCRIPTION
**Pull request by eumikhailov**
DataServiceClientRequestMessage contracts have changed (https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Client/Serialization/DataServiceClientRequestMessage.cs).
Sync it to https://github.com/OData/odata.net/blob/master/test/EndToEndTests/Tests/Client/Build.Desktop/TransportLayerTests/HttpClientRequestMessage.cs.

More details in issue https://github.com/OData/Extensions/issues/37

### CODE REVIEW

**Comment by eumikhailov** [comment]
Hello @chuanboz,
Please review

**Comment by eumikhailov** [comment]
Hello @xuzhg ,
Can you review this?

**Comment by xuzhg** [comment]
@groovyfeng

File: src/Microsoft.OData.Extensions.Client/Internals/Handlers/HttpClientRequestMessage.cs [hunk 1]
```diff
@@ -254,6 +299,18 @@ public override IODataResponseMessage EndGetResponse(IAsyncResult asyncResult)
         {
             return UnwrapAggregateException(() => new HttpClientResponseMessage(((Task<HttpResponseMessage>)asyncResult).Result, this.config));
         }
+        
+#if !PORTABLELIB && !(NETCOREAPP1_0 || NETCOREAPP2_0)
```
Comment (xuzhg) [inline src/Microsoft.OData.Extensions.Client/Internals/Handlers/HttpClientRequestMessage.cs:303]: remove the "PORTABLELIB"

Comment (eumikhailov) [reply src/Microsoft.OData.Extensions.Client/Internals/Handlers/HttpClientRequestMessage.cs]: removed

**Comment by eumikhailov** [comment]
Hello @xuzhg , please merge it if no admonition

**Pull request merged**